src/System.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs

```diff
@@ -32,6 +37,68 @@ public static bool AreConnStringsSetup()
             return !string.IsNullOrEmpty(NpConnStr) && !string.IsNullOrEmpty(TcpConnStr);
         }
 
+        public static bool IsServiceBrokerEnabled()
+        {
+            if (!serviceBrokerEnabled.HasValue)
+            {
+                serviceBrokerEnabled = false;
+                if (AreConnStringsSetup())
+                {
+                    try
+                    {
+                        var builder = new SqlConnectionStringBuilder(TcpConnStr);
+                        string database = builder.InitialCatalog;
+                        builder.ConnectTimeout = 2;
+                        using (var connection = new SqlConnection(builder.ToString()))
+                        using (var command = new SqlCommand("SELECT is_service_broker_enabled FROM sys.sys.databases WHERE name=@name", connection))
+                        {
+                            command.Parameters.AddWithValue("name", database);
+                            using (var reader = command.ExecuteReader())
+                            {
+                                if (reader.HasRows && reader.Read())
+                                {
+                                    serviceBrokerEnabled = (reader.GetInt32(0) == 1);
+                                }
+                            }
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        serviceBrokerEnabled = false;
+                    }
+                }
+            }
+            return serviceBrokerEnabled.Value;
+        }
+
+        public static bool IsDatabasePresent(string name)
+        {
+            if (databasesAvailable == null)
```

Comment:
Nit: Could be `databasesAvailable = databasesAvailable ?? new Dictionary<string, bool>();`